### DISCUSSION TITLE
It is not possible to member.remove_roles(member.roles) without removing the @everyone role from member.roles first

### DISCUSSION THREAD

**Discussion by IamMusavaRibica** [Rapptz/discord.py#5937] [Q&A] [answered]
## Summary
Wanting to remove all roles from a member makes an extra unnecessary step
I also included a workaround for this, and how this may be fixed
## Reproduction Steps
To remove all roles from a member:
`roles = member.roles`
`await member.remove_roles(*roles)`

## Expected Results

All roles that member had before, should be removed from him

## Actual Results

`Traceback (most recent call last):
  File "C:\Users\dario\AppData\Local\Programs\Python\Python37-32\lib\site-packages\discord\ext\commands\core.py", line 83, in wrapped
    ret = await coro(*args, **kwargs)
  File "C:\Users\dario\OneDrive\Radna površina\Focus Guild Bot rewrite\cogs\roleupdate.py", line 33, in fixroles
    await member.remove_roles(*roles)
  File "C:\Users\dario\AppData\Local\Programs\Python\Python37-32\lib\site-packages\discord\member.py", line 685, in remove_roles
    await req(guild_id, user_id, role.id, reason=reason)
  File "C:\Users\dario\AppData\Local\Programs\Python\Python37-32\lib\site-packages\discord\http.py", line 223, in request
    raise NotFound(r, data)
discord.errors.NotFound: 404 Not Found (error code: 10011): Unknown Role

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Users\dario\AppData\Local\Programs\Python\Python37-32\lib\site-packages\discord\ext\commands\bot.py", line 892, in invoke
    await ctx.command.invoke(ctx)
  File "C:\Users\dario\AppData\Local\Programs\Python\Python37-32\lib\site-packages\discord\ext\commands\core.py", line 797, in invoke
    await injected(*ctx.args, **ctx.kwargs)
  File "C:\Users\dario\AppData\Local\Programs\Python\Python37-32\lib\site-packages\discord\ext\commands\core.py", line 92, in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: NotFound: 404 Not Found (error code: 10011): Unknown Role`

However, I managed to avoid this by doing:
`everyone = guild.get_role(ROLE_ID)`
`roles = member.roles`
`roles.remove(everyone)`
`member.remove_roles(roles)`

How this might be able to be fixed:
When member.remove_roles is called, automatically remove the @everyone role from the roles passed, if there is one, because it cannot be removed and it throws an error

## Checklist

- [ ] I have searched the open issues for duplicates.
- [x] I have shown the entire traceback, if possible.
- [x] I have removed my token from display, if visible.
- [x] I have provided the intents that my bot is using.

## System Information

- Python v3.7.3-final
- discord.py v1.3.4-final
- aiohttp v3.6.2
- websockets v8.1
- system info: Windows 10 10.0.18362

**Comment by Rapptz** [comment, marked as answer]
In order to remove roles you should probably use `await member.edit(roles=[])`. `remove_roles` is a higher level function with atomicity but if you're clearing roles you should use the former.

**Reply by IamMusavaRibica** [reply]
Thank you, I tried to `await member.edit(roles=None` before and thought that this method cannot be used for removing roles

**Reply by MasterCoder343** [reply]
> In order to remove roles you should probably use `await member.edit(roles=[])`. `remove_roles` is a higher level function with atomicity but if you're clearing roles you should use the former.

nice but how will i return the roles when the user is unmuted

**Reply by ghost** [reply]
Thanks it helped!
I was designing a mute command

**Reply by Vertoxen** [reply]
> > In order to remove roles you should probably use `await member.edit(roles=[])`. `remove_roles` is a higher level function with atomicity but if you're clearing roles you should use the former.
> 
> nice but how will i return the roles when the user is unmuted

It's as easy as grabbing the member's roles in a variable and then editing the role again

**Reply by NimVrod** [reply]
> > In order to remove roles you should probably use `await member.edit(roles=[])`. `remove_roles` is a higher level function with atomicity but if you're clearing roles you should use the former.
> 
> nice but how will i return the roles when the user is unmuted

kinda of a late answer but if you still somehow didn't mange to do it or for someone else who is searching I did it like this::
``roles1 = member.roles #save the member roles``
``await member.edit(roles=[]) #remove all member roles``
``await member.edit(roles=roles1) #bring back the old roles ``

**Reply by cheemzboi** [reply]
i do this but its giving me this error now 
await member.edit(roles=[]) #remove all member roles
TypeError: Member.edit() missing 1 required positional argument: 'self'

**Comment by IamMusavaRibica** [comment]
i figured it out already , doesnt work if the member is a server booster
(discord doesnt let you remove that role)

ned, 16. svi 2021. 19:52 NimVrod ***@***.***> je napisao:
